File: flask_app/app/models/tier_benefits.py
from app.models.base import BaseModel, db
from app.constants import BenefitType

class TierBenefits(BaseModel):
    __tablename__ = 'tier_benefits'

    description = db.Column(db.Text, nullable=False)
    tier_id = db.Column(db.String(36), nullable=False)
    is_active = db.Column(db.Boolean, default=True)
    benefit_type  = db.Column(db.Enum(BenefitType), nullable=False)
    benefit_value = db.Column(db.JSON, nullable=False)
    sort_order = db.Column(db.Integer)
    
    def __repr__(self):
        return f"<TierBenefits {self.description}>"

File: java_membership_api/src/main/java/com/example/membership_api/model/TierBenefits.java
package com.example.membership_api.model;

import com.example.membership_api.constants.BenefitType;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Table(name = "tier_benefits")
@Data
@EqualsAndHashCode(callSuper = true)
public class TierBenefits extends BaseModel {

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(name = "tier_id", nullable = false, length = 36)
    private String tierId;

    @Column(name = "is_active")
    private boolean isActive = true;

    @Enumerated(EnumType.STRING)
    @Column(name = "benefit_type", nullable = false)
    private BenefitType benefitType;

    @Column(name = "benefit_value", columnDefinition = "JSON", nullable = false)
    private String benefitValue;

    @Column(name = "sort_order")
    private Integer sortOrder;
}
